apps/rand.c

```diff
@@ -198,7 +198,10 @@ int rand_main(int argc, char **argv)
     while (scaled_num > 0) {
         long chunk;
 
-        chunk = (scaled_num > buflen) ? buflen : scaled_num;
+        chunk = buflen;
+        if ((long)scaled_num < chunk)
```

Comment:
This does not seem to be right. Why not just `chunk = scaled_num > buflen ? buflen : (long)scaled_num;` ?

Reply:
because building with strict warnings was giving me a very odd compiler error on windows builds (potential loss of data by not casting the scaled_num to long to match the defintion of chunk).  Doint the cast then gave me an alternate warning about a sign comparison (uint64_t to long).  This avoids both warnings by setting chunk to buflen by default and scaling it back when scaled_num is less than that

Reply:
Hmm, but looking at the RAND_bytes() you should actually use `int chunk;` instead of long.

Reply:
I do not see how `chunk = scaled_num > buflen ? (int)buflen : (int)scaled_num;` and declaring chunk as int could produce any warnings.

Reply:
i can try it again, but it did.